Adding ExprLength's property expression

Romitou opened the pull request:
This Pull Request adds the expression property of the length of a string.
Example of script:
```applescript
on script load:
	set {_length} to "Hello."'s length
	print "%{_length}%" # Output: 6
```

If the `strValue.getSingle(ctx)` is null, I'd rather return 0 instead of empty array causing error.
I almost forgot, thanks to Olyno for his help!

Romitou committed:
:sparkles: Adding ExprLength's property expression

Olyno commented on src/main/java/io/github/syst3ms/skriptparser/expressions/ExprLength.java:
```diff
@@ -0,0 +1,50 @@
+ * Length of a string.
+ *
+ * @name Length
+ * @pattern length of %string%
```
Maybe add the ``%string%'s length`` pattern too
```suggestion
 * @pattern length of %string%
 * @pattern %string%'s length
```

Olyno reviewed (changes requested):
Good job and nice first pull request! Clean code! Maybe one thing is mising but that's all!

Romitou committed:
👌 Adding a pattern

Co-authored-by: Olyno <alexlew95@gmail.com>

Syst3ms reviewed (changes requested):
This should be extending the PropertyExpression class.

Romitou committed:
👌 Extend PropertyExpression

Syst3ms reviewed (changes requested):
It's not all to just extend the class, this particular use case could get away with only extending getPropertyFunction() and toString(). The other methods' (init, getValues) default functionalities are perfectly fine for this purpose.

Mwexim commented:
This would be the perfect example of a property to have in the parser. If needed, I can make the correct class since I am using the new PropertyExpression for quite some time now.

Mwexim commented:
To add to the above: below is the provided file from my implementation. Hopefully it makes clear how to use the property expression!
https://github.com/Mwexim/Buzzle/blob/master/src/main/java/com/github/mwexim/buzzle/expressions/ExprStringLength.java
The only thing that’s wrong there is the redundant check for the array length in the getPropertyFunction() method as the PropertyExpression class handles that.

Romitou commented:
Thank you, this helped me a lot. Thanks to Syst3ms for its help too!

We were indeed checking the length of the array in the getPropertyFunction(); which I removed following your remark because the getValues() handles it. So I corrected everything on my side, hoping it's ok now!

Romitou committed:
👌 Changes due to code review

Mwexim reviewed (approved):
Great implementation. I think everything is okay as far as I can see.

Syst3ms commented on src/main/java/io/github/syst3ms/skriptparser/expressions/ExprLength.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Override
+    public Function<String[], Number[]> getPropertyFunction() {
+        return strings -> new Number[] {strings[0].length()};
```
A alright, the last change I'll request is simply some ""null safety"". By that I mean that `getValues()` (which is fed into this function) may be empty.
I'll say again the fundamental equivalence in this parser : `getSingle() == null` implies that `getValues().length == 0` and vice-versa.
```suggestion
        return strings -> strings.length == 0 ? new Number[0] : new Number[]{strings[0].length()};
```

Romitou replied:
All right, thank you!

Syst3ms reviewed (changes requested):
Alright, only some minor changes left now, it should be all good after this.

Romitou committed:
👌 Adding some null safety checks

Co-authored-by: Syst3ms <arthmm14@gmail.com>

Mwexim commented:
@Syst3ms isn’t the empty array check redundant if you know that PropertyExpression automatically returns an empty array on its own when that happens?

If you want the functionality where it returns 0 when the string is not set, you will need to override the #getValues(ctx) method as it will handle this itself for the above reasons. 

Either way, if that doesn’t matter, the redundant check should be removed.

Syst3ms commented:
You are correct, I actually forgot how it was implemented. I'll fix that and add some javadoc to clear things up in the future.
